extensions/aris/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,493 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as nls from 'vscode-nls';
+import * as azdata from 'azdata';
+import { IControllerError, IEndPointsResponse } from '../controller/types';
+import { ArisController } from '../controller/controller';
+
+const localize = nls.loadMessageBundle();
+
+export class AddControllerDialog {
+
+	private readonly DialogTitle: string = localize('bigDataClusters.addControllerDialog.title', 'Add New Controller');
+	private readonly SignInButtonText: string = localize('bigDataClusters.addControllerDialog.ok.title', 'Add');
+	private readonly CancelButtonText: string = localize('bigDataClusters.addControllerDialog.cancel.title', 'Cancel');
+
+	private dialog: azdata.window.Dialog;
+	private modelBuilder: azdata.ModelBuilder;
+
+	private urlInputBox: azdata.InputBoxComponent;
+	private usernameInputBox: azdata.InputBoxComponent;
+	private passwordInputBox: azdata.InputBoxComponent;
+	private rememberPwCheckBox: azdata.CheckBoxComponent;
+	private errorTextContainer: azdata.FlexContainer;
+	private callback: (e: IEndPointsResponse, rememberPassword: boolean) => any;
+	private callbackCancel: () => any;
+	private callbackError: (e: IControllerError) => any;
+
+	private prefilledValues: {
+		url?: string,
+		username?: string,
+		password?: string,
+		rememberPassword?: boolean,
+	} = {};
+
+	public setPrefilledValues(prefilledValues: {
+		url?: string,
+		username?: string,
+		password?: string,
+		rememberPassword?: boolean,
+	}): void {
+		if (!prefilledValues || Object.keys(prefilledValues).length === 0) {
+			return;
+		}
+		this.prefilledValues.url = prefilledValues.url;
+		this.prefilledValues.username = prefilledValues.username;
+		this.prefilledValues.password = prefilledValues.password;
+		this.prefilledValues.rememberPassword = prefilledValues.rememberPassword;
+	}
+
+	constructor(prefilledValues?: {
+		url?: string,
+		username?: string,
+		password?: string,
+		rememberPassword?: boolean,
+	}) {
+		this.setPrefilledValues(prefilledValues);
+	}
+
+	public async showDialog(
+		callback: (e: IEndPointsResponse, rememberPassword: boolean) => any,
+		callbackCancel?: () => any,
+		callbackError?: (e: IControllerError) => any,
+	): Promise<void> {
```

Comment:
As mentioned this is a void method, remove `async` and make it void.

Reply:
Please note that this pattern is what I followed in CMS. Fixed.